pkg/corerp/frontend/controller/applications/graph_util.go

```diff
@@ -547,61 +556,62 @@ func findSourceResource(source string, allResources []generated.GenericResource)
 	return orig, ErrInvalidSource
 }
 
-// providesFromAPIData is specifically to support HTTPRoute.
-func providesFromAPIData(resource generated.GenericResource) []*corerpv20231001preview.ApplicationGraphConnection {
-	// Any Radius resource type that exposes a port uses the following property path to return them.
-	// The port may have a 'provides' attribute that specifies a httproute.
-	// This route should be parsed to find the connections between containers.
-	// For example, if container A provides a route and container B consumes it,
-	// then we have port.provides in container A and container.connection in container B.
-	// This gives us the connection: container A --> route R --> container B.
-	// Without parsing the 'provides' attribute, we would miss the connection between container A and route R.
-
-	p, err := jsonpointer.New(portsPath)
-	if err != nil {
-		// This should never fail since we're hard-coding the path.
-		panic("parsing JSON pointer should not fail: " + err.Error())
+// connectionsResolver resolves the outbound connections for a resource from the generic resource representation.
+// For example if the resource is an 'Applications.Core/container' then this function can find it's connections
+// to other resources like databases. Conversely if the resource is a database then this function
+// will not find any connections (because they are inbound). Inbound connections are processed later.
+func connectionsResolver(resources []generated.GenericResource) resolver {
```

Comment:
Not relevant for this review, but long term I'd like to make this behavior consistent across APIs by using Capabilities + Well-Known API shapes, or Well-Known Child resources.

Consider how we'd make this extensible 👍 When users can define their own types, they should be able to support all of the features of the built-in ones.

Reply:
Basically anyplace we write code that handles a specific resource type like `Applications.Core/gateways`, we might need to reconsider in the future.

Reply:
I intentionally didn't try to refactor it further to keep the change short. Instead of strongly typed one, I would try to using jsonpointer to look up the `connections.destination`. We can support multiple types. Also for custom resource, we can ask contributors to use the specific connection properties if they want to use appgraph. If you like this idea, I can make the change now.

Reply:
I don't think we should change it now - we should discuss during the design of user-defined-types. I just wanted to mention the idea.